packages/js/src/executors/copy-workspace-modules/copy-workspace-modules.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  type ExecutorContext,
+  logger,
+  parseTargetString,
+  type ProjectGraph,
+  readJsonFile,
+  workspaceRoot,
+} from '@nx/devkit';
+import { interpolate } from 'nx/src/tasks-runner/utils';
+import { type CopyWorkspaceModulesOptions } from './schema';
+import { cpSync, existsSync, mkdirSync, writeFileSync } from 'node:fs';
+import { dirname, join } from 'path';
+import { lstatSync } from 'fs';
+
+export default async function copyWorkspaceModules(
+  schema: CopyWorkspaceModulesOptions,
+  context: ExecutorContext
+) {
+  logger.log('Copying Workspace Modules to Build Directory...');
+  const outputDirectory = getOutputDir(schema, context);
+  const packageJson = getPackageJson(schema, context);
+  createWorkspaceModules(outputDirectory);
+  handleWorkspaceModules(outputDirectory, packageJson, context.projectGraph);
+  logger.log('Success!');
+  return { success: true };
+}
+
+function handleWorkspaceModules(
+  outputDirectory: string,
+  packageJson: {
+    dependencies?: Record<string, string>;
+  },
+  projectGraph: ProjectGraph
+) {
+  if (!packageJson.dependencies) {
+    return;
+  }
+
+  for (const [pkgName, pkgVersion] of Object.entries(
+    packageJson.dependencies
+  )) {
+    if (pkgVersion.startsWith('workspace:') || pkgVersion.startsWith('file:')) {
```

Comment:
What if users use `*`? Considering NPM doesn't support `workspace:` I think it's down to `*` or `file:`. Or perhaps the version is valid semver, e.g. `1.0.0`, and it just matches what's in the monorepo.